docs/analyze.md

```diff
@@ -1,124 +1,141 @@
+
+  <!-- Configurations for DataCollectors -->  
+  <DataCollectionRunSettings>  
+    <DataCollectors>  
+      <DataCollector friendlyName="MyDataCollector" uri="datacollector://MyCompany/MyDataCollector/1.0">  
+        <Configuration>
+                    <LogFileName>DataCollectorLogs.txt</LogFileName>
+        </Configuration>  
+      </DataCollector>
+    </DataCollectors>  
+  </DataCollectionRunSettings>  
+</RunSettings>
+```
+Below is the sample command for enabling DataCollectors using runsettings
+```
+> vstest.console.exe test_project.dll /settings:datacollection.runsettings
+```
+### Using vstest.console args
+DataCollectors can be configured and used through first class command line arguments `/collect` and `/testadapterpath`. Hence, for common DataCollection scenarios, separate runsettings file may not be required.
+
+Below is the sample command to configure and use DataCollectors through vstest.console command line.
+```
+> vstest.console.exe test_project.dll /collect:"Code Coverage"
+> vstest.console.exe test_project.dll /collect:"MyDataCollector" /testadapterpath:<Path to MyDataCollector assembly>
+```
+Please note that `testadapterpath` is not required for DataCollectors shipped along with TPv2.
+
+### Using TestSettings
+While the recommended way is to use [runsettings](#Using-RunSettings) or [vstest.console args](#Using-vstest.console-args), there are few DataCollectors which only worked with testsettings.
+E.g.: `System Information` DataCollector. Below is the sample testsettings for using `System Information` DataCollctor.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<TestSettings name="TestSettings1" id="2d572055-54c0-4cac-8a55-9d7ffb48ac17" xmlns="http://microsoft.com/schemas/VisualStudio/TeamTest/2010">
+  <Description>These are default test settings for a local test run.</Description>
+  <Deployment enabled="false" />
+  <Execution>
+    <AgentRule name="LocalMachineDefaultRole">
+      <DataCollectors>
+        <DataCollector uri="datacollector://microsoft/SystemInfo/1.0" assemblyQualifiedName="Microsoft.VisualStudio.TestTools.DataCollection.SystemInfo.SystemInfoDataCollector, Microsoft.VisualStudio.TestTools.DataCollection.SystemInfo, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" friendlyName="System Information">
+        </DataCollector>
+      </DataCollectors>
+    </AgentRule>
+  </Execution>
+  <Properties />
+</TestSettings>
+```
+Below is the sample command for enabling DataCollectors using testsettings
+```
+> vstest.console.exe test_project.dll /settings:datacollection.testsettings
+```
+### Enable/Disable a DataCollector
+All DataCollectors configured in the .runsettings files are loaded automatically and are enabled to participate for run, unless explicitely disabled using boolean valued attribute named `enabled`.
+
+For example, only `coverage` DataCollector will be enabled for a test run with
 below runsettings:
 
 ```xml
 <DataCollectionRunSettings> 
```

Comment:
pl remove code coverage partial section of runsettings. it has incorrect friendly name. also people may make mistake to add this section only as runsettings.